graders/c/cgrader/cgrader.py

```diff
@@ -5,23 +5,21 @@
 import re
 import shlex
 import subprocess
-import sys
 import tempfile
-import threading
 
 import lxml.etree as ET
 
 CODEBASE = "/grade/student"
 DATAFILE = "/grade/data/data.json"
 SB_USER = "sbuser"
 
-TIMEOUT_MESSAGE = (
-    "\n\nTIMEOUT! Typically this means the program took too long,"
-    + "\nrequested more inputs than provided, or an infinite loop was found."
-    + "\nIf your program is reading data using scanf inside a loop, this "
-    + "\ncould also mean that scanf does not support the input provided "
-    + "\n(e.g., reading an int if the input is a double).\n"
-)
+TIMEOUT_MESSAGE = """
+TIMEOUT! Typically this means the program took too long,
```

Comment:
The original had two line breaks at the start. There is a reason for this: if there is a message printed before timeout happens, you want a space between the message and the timeout error.
```suggestion
TIMEOUT_MESSAGE = """

TIMEOUT! Typically this means the program took too long,
```